packages/steps/src/__tests__/BuildWorkflowValidator-test.ts

```diff
@@ -343,4 +344,35 @@ describe(BuildWorkflowValidator, () => {
       'Input parameter "input2" for step "test3" uses an expression that references an undefined output parameter "output2" from step "test2".'
     );
   });
+  test('unallowed platform for build step', async () => {
+    const id = 'test';
+    const displayName = BuildStep.getDisplayName({ id });
+    const fn: BuildStepFunction = () => {};
+
+    const ctx = createMockContext({ runtimePlatform: BuildPlatform.LINUX });
+    const workflow = new BuildWorkflow(ctx, {
+      buildSteps: [
+        new BuildStep(ctx, {
+          id,
+          displayName,
+          fn,
+          workingDirectory: ctx.workingDirectory,
+          allowedPlatforms: [BuildPlatform.DARWIN],
+        }),
+      ],
+      buildFunctions: {},
+    });
+
+    const validator = new BuildWorkflowValidator(workflow);
+    const error = getError<BuildWorkflowError>(() => {
+      validator.validate();
+    });
+    expect(error).toBeInstanceOf(BuildWorkflowError);
+    assert(error instanceof BuildWorkflowError);
```

Comment:
Isn't this assertion duplicated?

Reply:
`error` could be falsy - check the `getError` signature. We have this `assert` so we don't have to use optional chaining in `expect`s below.  Keep in mind that the `expect` above doesn't guarantee that `error` is actually defined.